custom_model_runner/datarobot_drum/drum/language_predictors/base_language_predictor.py

```diff
@@ -174,6 +197,94 @@ def _predict(self, **kwargs) -> RawPredictResponse:
         """Predict on input_filename or binary_data"""
         pass
 
+    def chat(self, completion_create_params):
+        start_time = time.time()
+        try:
+            response = self._chat(completion_create_params)
+            response = self._validate_chat_response(response)
+        except Exception as e:
+            self._mlops_report_error(start_time)
+            raise e
+
+        if not is_streaming_response(response):
+            self._mlops_report_chat_prediction(
+                completion_create_params, start_time, response.choices[0].message.content
+            )
+            return response
+        else:
+
+            def generator():
+                message_content = ""
+                try:
+                    for chunk in response:
+                        message_content += (
+                            chunk.choices[0].delta.content
+                            if chunk.choices and chunk.choices[0].delta.content
+                            else ""
+                        )
+                        yield chunk
+                except Exception:
+                    self._mlops_report_error(start_time)
+                    raise
+
+                self._mlops_report_chat_prediction(
+                    completion_create_params, start_time, message_content
+                )
+
+            return generator()
+
+    def _chat(self, completion_create_params):
+        raise NotImplementedError("Chat is not implemented ")
+
+    def _mlops_report_chat_prediction(self, completion_create_params, start_time, message_content):
+        execution_time_ms = (time.time() - start_time) * 1000
+
+        self._mlops.report_deployment_stats(num_predictions=1, execution_time_ms=execution_time_ms)
+
+        latest_message = completion_create_params["messages"][-1]["content"]
+        features_df = pd.DataFrame([{"prompt": latest_message}])
+
+        predictions = [message_content]
+        try:
+            self._mlops.report_predictions_data(
+                features_df,
+                predictions,
+                association_ids=[str(uuid.uuid4())],
```

Comment:
Is this by design to have a random uuid?

Reply:
Yes. I need to pass some unique association id. This id should be passed to moderations framework when their chat integration is implemented at some point